### DISCUSSION TITLE
How to integrate metabase dashboard code to directus

### DISCUSSION THREAD

**Discussion by Jeyakumarrathnaselvi** [directus/directus#15615] [Archive]
How to integrate Metabase dashboard code to directus

![image](https://user-images.githubusercontent.com/36895131/190700135-0cb52584-0f86-46d5-a862-68268f081c08.png)

1. Compiled a extension module using npm extension build

![image](https://user-images.githubusercontent.com/36895131/190840539-fa510167-7235-4bc8-a8dc-41c38667e7bf.png)

2. Updated Vue Js and index.js with the metabase code like below

![image](https://user-images.githubusercontent.com/36895131/190840596-f4ade6eb-5fb8-4cab-ad23-c5b3e624c7b4.png)

![image](https://user-images.githubusercontent.com/36895131/190840605-121bb9b9-c0e4-4aeb-ac7b-29c59249aad8.png)

3. then again compiled the module using 

![image](https://user-images.githubusercontent.com/36895131/190840637-75f21db8-bf65-4323-9b7a-62dcf76ebfee.png)

4. Then uploaded index,js file to respective extension folder

![image](https://user-images.githubusercontent.com/36895131/190840693-65ef3ccb-1925-4452-86cc-8343eb91b2f1.png)

5. Validated directus log if any error but not getting any error

![image](https://user-images.githubusercontent.com/36895131/190840759-c3b2aa12-0b6f-4b55-897f-b3183240848e.png)

I supposed to get the icon in my left side menu correct but not getting it

**Comment by azrikahar** [comment]
Granted I don't have personal experience embedding Metabase, I believe the **_server_** side code snippet is not 100% complete, in the sense that you will need to handle the returning of the formed `iframeUrl`. 

A Directus **module** custom extension is only App-side, not API or server side, so we can't mix them that way. With the information at hand, I think you will need to create a custom **endpoint** extension to return the `iframeUrl`, then use it in your custom **module** extension as the src of the iframe.

**Reply by Jeyakumarrathnaselvi** [reply]
Thanks but i am not a coder so my question may be very basic. Sorry for this. i created a endpoint like this
![image](https://user-images.githubusercontent.com/36895131/191079148-4c82a3c6-9a73-4a6a-b7cb-b774253fffaa.png)

and uploaded to index.js to extension endpoint folder but getting this error

![image](https://user-images.githubusercontent.com/36895131/191079511-1abbb9a3-0ee2-40b0-a164-1e23a2ac44cd.png)

**Reply by azrikahar** [reply]
>  i created a endpoint like this

Line 6 to line 18 should be nested in the `router.get('/')` handler, and replace `res.send('Hello, World!')` with something like `res.json({iframeUrl})` to send it as a json (feel free to modify this if that's not what you prefer) 👍

> but getting this error

That is because `jsonwebtoken` needs to be installed in your Directus project. You can refer to https://www.npmjs.com/package/jsonwebtoken#install for the installation step.

**Comment by Powma** [comment]
Hi,

Your private key Metabase isn't hide. Maybe it's ok for you, but I prefer to notify you.
